tests/integration/components/file-upload-test.js

```diff
@@ -0,0 +1,37 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { upload } from 'ember-file-upload/test-support';
+
+module('Integration | Component | file-upload', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('uploading a file calls onfileadd', async function(assert) {
+    this.setFile = (file) => {
+      assert.step(file.name);
+    };
+
+    await render(hbs`{{file-upload name="test" onfileadd=(action this.setFile)}}`);
+
+    await upload('input[type="file"]', new File([], 'dingus.txt'));
+
+    assert.verifySteps(['dingus.txt']);
+  });
+
+  test('uploading multiple files calls onfileadd multiple times', async function(assert) {
+    this.setFile = (file) => {
+      assert.step(file.name);
+    };
+
+    await render(hbs`{{file-upload name="test" onfileadd=(action this.setFile)}}`);
+
+    await upload(
+      'input[type="file"]',
+      new File([], 'dingus.txt'),
+      new File([], 'dingus.png')
```

Comment:
Should these be in an array? Isn't that what the test helper uses?

Reply:
It uses the [rest parameter syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameter):

```js
export async function upload(selector, ...files) {
}
```

An awesome design choice IMO. It does not alter the existing interface and does not allow multiple types for the same argument. This should make documentation straight forward and even tooling support (if we provides types as some point of time).